### ISSUE TITLE
Missing Types

### ISSUE THREAD

**Issue by treedbox** [vladmandic/face-api#27]
Hi Vladimir,
Despite works normally, your module is missing the types. When I try to use in the VS Code, this message is shown when we hover the import:
```
Could not find a declaration file for module '@vladmandic/face-api'.
'.../node_modules/@vladmandic/face-api/dist/face-api.node.js' implicitly has an 'any' type.
Try `npm i --save-dev @types/vladmandic__face-api`
 if it exists or add a new declaration (.d.ts) file containing `declare module '@vladmandic/face-api';`js(7016)
```

`import * faceapi from "@vladmandic/face-api";`
or:
`import * as faceapi2 from '@vladmandic/face-api/dist/face-api.esm.js';`

The original face-api.js works.

Later, I plan to clone your project locally to see what we can do.

**Comment by vladmandic** [comment]
It depends on your VSCode configuration. But yes, there are no explict types generated since I switched build to got directly from TypeScript to ESM. I'll take a look to add build step to generate types without the need to do TS->JS compile.

**Comment by vladmandic** [comment]
I've added build step to create typings in `dist/face-api.d.ts`.  
new version is published.

**Comment by vladmandic** [comment]
@treedbox can you confirm that typings now work?

**Comment by volskaya** [comment]
@vladmandic Don't work for me - `File '....../face-api.d.ts' is not a module.`

**Comment by vladmandic** [comment]
my bad, should be fixed now.

**Comment by volskaya** [comment]
Now its throwing these everywhere the types try to import `tfjs.esm.js`

```
......../face-api/types/index.d.ts:1:21 - error TS7016: Could not find a declaration file for module '../dist/tfjs.esm.js'. '........face-api/dist/tfjs.esm.js' implicitly has an 'any' type.

1 import * as tf from '../dist/tfjs.esm.js';
```

**Comment by vladmandic** [comment]
@volskaya can you share your project configuration? the type checking for embedded tfjs should be skipped.

**Comment by volskaya** [comment]
This one isn't thrown by the linter, it's thrown by the typescript builder. The linter doesn't show any errors, nor warnings.

I already wrote an abstraction for my simple use case and this face-api port is working flawlessly, after I replace the TS imports with JS requires. 👍

I can't share my project, but it's the default firebase cloudfunctions configuration that you create with `firebase init`.

**Comment by vladmandic** [comment]
hmm...can you share just your `tsconfig.json` as that is used by vscode typescript builder?

**Comment by volskaya** [comment]
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "outDir": "lib",
    "sourceMap": true,
    "strict": true,
    "target": "es2019",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "resolveJsonModule": true,
    "strictNullChecks": false
  },
  "compileOnSave": true,
  "include": ["src"]
}
```

**Comment by vladmandic** [comment]
ok, i've reproducted it, but - there is no easy solution:  

- `tfjs.esm` is a virtual package which means it binds to different version of tfjs depending on platform (browser, node, node-gpu), so i can't create single typings for it. that's a bad side of npm, there is no elegant way to deal with multiple build targets inside same project. you can create `@node_modules/@vladmandic/face-api/dist/tfjs.esm.js` with a single line `export * from '@tensorflow/tfjs';` (and change depending if you're using browser or node) and that fixes 99% of errors, but not all.
- i might work harder on this **if** @tensorflow/tfjs itself was up-to-date with typings. unfortunately, it has several well-known issues that are planned to be addressed in tfjs 3.0 later this year - for example, complete mismatch of typings for webgl.

as it is, recommendation for @tensorflow/tfjs is to have `"skipLibCheck": true"` in `tsconfig.json` - once that is added, typescript host does not check validity of exports from libraries (it still uses typings for hints, so you don't loose type checks).  
and same recommendation stands for `face-api` - please set `"skipLibCheck": true"`. i hope to resolve this issue once tfjs typings themselves are fixed.

**Comment by treedbox** [comment]
@vladmandic Now it's working like a charm :D
Installed as devDependencies:
`npm i -D @vladmandic/face-api`

index.ts:
```ts
import * as faceapi from "@vladmandic/face-api";
console.log("faceapi:", faceapi);
console.log("faceapi.ssdMobilenetv1:", faceapi.ssdMobilenetv1);
```
![Screenshot from 2020-12-24 16-00-54](https://user-images.githubusercontent.com/4041664/103103651-44e98b80-4601-11eb-8d9c-1501b4005ff6.png)


![Screenshot from 2020-12-24 15-58-33](https://user-images.githubusercontent.com/4041664/103103621-0fdd3900-4601-11eb-8074-8d1e82921bd6.png)

Thanks! Until next year!
